metadata/buckets.go

```diff
@@ -182,6 +182,41 @@ func createBucketIfNotExists(tx *bolt.Tx, keys ...[]byte) (*bolt.Bucket, error)
 	return bkt, nil
 }
 
+func namespacesBucketPath() []byte {
+	return bucketKeyVersion
+}
+
+func getNamespacesBucket(tx *bolt.Tx) *bolt.Bucket {
+	return getBucket(tx, namespacesBucketPath())
+}
+
+func getSharableBucket(tx *bolt.Tx, dgst digest.Digest) *bolt.Bucket {
+	var bkt *bolt.Bucket
+	nsbkt := getNamespacesBucket(tx)
+	cur := nsbkt.Cursor()
+	for k, v := cur.First(); k != nil; k, v = cur.Next() {
+		// Check if v has shared values
+		// It looks weird but go will complain that v is
+		// never used if this is not here.
+		_ = v
+		labelsBkt := getNamespaceLabelsBucket(tx, string(k))
+		if labelsBkt == nil {
+			continue
+		}
+		labels := map[string]string{}
+		labelsBkt.ForEach(func(k, v []byte) error {
+			labels[string(k)] = string(v)
+			return nil
+		})
+		// if it is, check if K has image.
+		if _, ok := labels[sharedNamespaceLabel]; ok {
+			bkt = getBlobBucket(tx, string(k), dgst)
+			break
+		}
```

Comment:
I think we can at least stop ForEach() when we find shardNamespaceLabel in it.

Reply:
Returning a custom error to stop the loop would work.

Reply:
```go
var foundSharedLabel = errors.New("found shared label")
```

```go
err := labelsBkt.ForEach(func(k, _ []byte) error {
	if string(k) == sharedNamespaceLabel {
		sharedBucket = string(k)
		return foundSharedLabel
	}
})
if err == foundSharedLabel {
	bkt = getBlobBucket(tx, string(k), dgst)
	break
}
```